pkg/controller/clusterconfig/migration.go

```diff
@@ -111,7 +111,7 @@ func (r *ReconcileClusterConfig) prepareOperatorConfigForNetworkTypeMigration(ct
 		}
 		return nil
 	}
-	if !mtuApplied && cniReady {
+	if !mtuApplied && cniReady && operConfig.Spec.DefaultNetwork.Type != operv1.NetworkType(clusterConfig.Spec.NetworkType) {
 		if isRollback {
```

Comment:
Should this if block happen after mtuMigration retrieval? Currently if calculateRoutableMTU fails we will not re-enter the step 2 block on rollback.

Reply:
Good point. Fixed.